File: ruby/04.rb
#!/usr/bin/env ruby

require 'httparty'

class GitHubber
  def self.events
    stream = HTTParty.get 'https://api.github.com/events'
    stream[0,10].map do |event|
      {
        id: event['id'],
        created: event['created_at'],
        repo: event['repo']['name'],
        author: event['actor']['login'],
        commits: event['payload'].count
      }
    end
  end
end

if __FILE__ == $0
  puts GitHubber.events.map(&:inspect).join("\n")
end

File: julia/04.jl
#!/usr/bin/env julia

import Requests
import JSON

response = Requests.get("https://api.github.com/events")

# stuck; any hit to an HTTPS URL returns
  # ERROR: stream is closed or unusable
  #  in check_open at stream.jl:294
  #  in write at stream.jl:730

#data = JSON.parse(response.data[1:10])
#println(data)


